src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs

```diff
@@ -1100,7 +1134,7 @@ internal override bool IsRecordDeclaration(SyntaxNode node)
             => node.IsKind(SyntaxKind.RecordDeclaration, SyntaxKind.RecordStructDeclaration);
 
         internal override SyntaxNode? TryGetContainingTypeDeclaration(SyntaxNode node)
-            => node.Parent!.FirstAncestorOrSelf<BaseTypeDeclarationSyntax>();
+            => node is CompilationUnitSyntax ? null : node.Parent!.FirstAncestorOrSelf<BaseTypeDeclarationSyntax>();
```

Comment:
>node is CompilationUnitSyntax ? null : node.Parent! [](http://example.com/codeflow?start=15&length=51)

Should this be `node.Parent?.` instead, to be more forgiving?

Reply:
This is called to get the containing type for a member that has a lambda, so global statements is the only scenario where there wouldn't be a parent.

Reply:
Ahhh I totally missed that! Thank you